ports/rp2/mpconfigport.h

```diff
@@ -71,8 +71,19 @@
 #define MICROPY_CONFIG_ROM_LEVEL                (MICROPY_CONFIG_ROM_LEVEL_EXTRA_FEATURES)
 #endif
 
+#ifndef MICROPY_HW_ENABLE_PSRAM
+#define MICROPY_HW_ENABLE_PSRAM (0)
+#endif
+
 // Memory allocation policies
+#if MICROPY_HW_ENABLE_PSRAM
+#define MICROPY_GC_STACK_ENTRY_TYPE             uint32_t
+#else
 #define MICROPY_GC_STACK_ENTRY_TYPE             uint16_t
+#endif
+#ifndef MICROPY_GC_SPLIT_HEAP
+#define MICROPY_GC_SPLIT_HEAP                   (0) // whether PSRAM is added to or replaces the heap
```

Comment:
@Gadgetoid I understand you previously had a problem using split heap where it would crash/hang when collecting. However I've been tinkering around with the split heap and have not had any trouble (both manually forcing collection, and allowing automatic collection with the threshold set to both -1 and some non-zero number).

Could you test again? If you can still cause it to crash/hang, could you share more about your test?

If the problem has been resolved, I think it would be ideal to use the split heap by default. An extra ~500kB is always nice to have!

Reply:
Agreed. We've got split heap enabled on our PPP2 and PPP2W boards which have been in customers hands for some time now. And some builds of PGA2350 too. I think it's safe to say we can (and should) default this to on. Added to my TODO list!